internal/router/api/v1/domain.go

```diff
@@ -119,6 +120,40 @@ func PostDomains(c *gin.Context) {
 			return
 		}
 		domainListResponse.Domains = domains
+
+	case "azure":
+		if domainListRequest.AzureAuth.ClientID == "" ||
+			domainListRequest.AzureAuth.ClientSecret == "" ||
+			domainListRequest.AzureAuth.SubscriptionID == "" ||
+			domainListRequest.AzureAuth.TenantID == "" {
```

Comment:
To prevent these long ass checks, you might want to add a method to `domainListRequest` that returns true or false if these fields are empty, so you don't have to continue writing it like such.